tests/unit/test_charm.py

```diff
@@ -724,3 +724,145 @@ def test_get_rgw_endpoints_action_after_traefik_is_integrated(
+            f"--stop-osds={stop_osds}",
+            f"--force={force}",
+            f"--dry-run={dry_run}",
+        ]
+        self.harness.charm._enter_maintenance_action(action_event)
+        subprocess_run.assert_called_with(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True,
+            timeout=180,
+        )
+        action_event.set_results.assert_called_with(
+            {"status": "success", "output": mock_process.stdout}
+        )
+
+    @patch("microceph.subprocess.run")
+    def test_enter_maintenance_action_fail(self, subprocess_run):
+        name = "node-1"
+        force = False
+        dry_run = False
+        set_noout = True
+        stop_osds = False
+        action_event = MagicMock()
+        action_event.params = {
+            "name": name,
+            "set-noout": set_noout,
+            "stop-osds": stop_osds,
+            "dry-run": dry_run,
+            "force": force,
+        }
+        cmd = (
+            [
+                "sudo",
+                "microceph",
+                "cluster",
+                "maintenance",
+                "enter",
+                "node-1",
+                f"--set-noout={set_noout}",
+                f"--stop-osds={stop_osds}",
+                f"--force={force}",
+                f"--dry-run={dry_run}",
+            ],
+        )
+        subprocess_run.side_effect = CalledProcessError(returncode=1, cmd=cmd, stderr="some error")
+        self.harness.charm._enter_maintenance_action(action_event)
+        action_event.set_results.assert_called_with(
+            {"status": "failure", "reason": f"Command '{cmd}' returned non-zero exit status 1."}
+        )
+        action_event.fail.assert_called()
+
+    @patch("microceph.subprocess.run")
+    def test_exit_maintenance_action_succeed(self, subprocess_run):
+        output = "plan or error message"
+        mock_process = MagicMock()
+        mock_process.stdout = PropertyMock(return_value=output)
+
+        subprocess_run.return_value = mock_process
+        action_event = MagicMock()
+        name = "node-1"
+        dry_run = False
+        action_event.params = {
+            "name": name,
+            "dry-run": dry_run,
+        }
+        cmd = [
+            "sudo",
+            "microceph",
+            "cluster",
+            "maintenance",
+            "exit",
+            "node-1",
+            f"--dry-run={dry_run}",
+        ]
+        self.harness.charm._exit_maintenance_action(action_event)
+        subprocess_run.assert_called_with(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True,
+            timeout=180,
+        )
+        action_event.set_results.assert_called_with(
+            {"status": "success", "output": mock_process.stdout}
+        )
+
+    @patch("microceph.subprocess.run")
+    def test_exit_maintenance_action_fail(self, subprocess_run):
+        name = "node-1"
+        dry_run = False
```

Comment:
The default for dry-run is false already though?

Reply:
This is not relevant anymore in the latest commits